include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,349 @@
+ * @ingroup senss
+ * @defgroup senss_datatypes Data Types
+ * @ingroup senss
+ */
+
+#include <zephyr/senss/senss_datatypes.h>
+#include <zephyr/senss/senss_sensor_types.h>
+
+/**
+ * @brief Sensor Subsystem API
+ * @addtogroup senss_api
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @struct senss_sensor_version
+ * @brief Sensor Version
+ */
+struct senss_sensor_version {
+	union {
+		uint32_t value;
+		struct {
+			uint8_t major;
+			uint8_t minor;
+			uint8_t hotfix;
+			uint8_t build;
+		};
+	};
+};
+
+#define SENSS_SENSOR_INVALID_HANDLE (-1)
+#define SENSS_INDEX_ALL (-1)
+
+#define SENSS_SENSOR_VERSION(_major, _minor, _hotfix, _build)           \
+	((_major) << 24 | (_minor) << 16 | (_hotfix) << 8 | (_build))
+
+
+/**
+ * @brief Sensor flag indicating if this sensor is on event reporting data.
+ *
+ * Reporting sensor data when the sensor event occurs, such as a motion detect sensor reporting
+ * a motion or motionless detected event.
+ */
+#define SENSS_SENSOR_FLAG_REPORT_ON_EVENT			BIT(0)
+
+/**
+ * @brief Sensor flag indicating if this sensor is on change reporting data.
+ *
+ * Reporting sensor data when the sensor data changes.
+ *
+ * Exclusive with  \ref SENSS_SENSOR_FLAG_REPORT_ON_EVENT
+ */
+#define SENSS_SENSOR_FLAG_REPORT_ON_CHANGE			BIT(1)
+
+
+/**
+ * @brief Sensor subsytem sensor state.
+ *
+ */
+enum senss_sensor_state {
+	SENSS_SENSOR_STATE_NOT_READY = 1,
+	SENSS_SENSOR_STATE_READY = 2,
+};
+
+/**
+ * @brief Sensor subsystem sensor config attribute
+ *
+ */
+enum senss_sensor_attribute {
+	SENSS_SENSOR_ATTRIBUTE_INTERVAL = 0,
+	SENSS_SENSOR_ATTRIBUTE_SENSITIVITY = 1,
+	SENSS_SENSOR_ATTRIBUTE_LATENCY = 2,
+	SENSS_SENSOR_ATTRIBUTE_MAX,
+};
+
+
+/**
+ * @brief Sensor data event receive callback.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer with sensor data.
+ *
+ * @param size The buffer size in bytes.
+ */
+typedef void (*senss_data_event_t)(
+		int handle,
+		void *buf, int size);
+
+/**
+ * @brief Sensor bias after calibartion event receive callback.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer with bias data.
+ */
+typedef void (*senss_bias_event_t)(
+		int handle,
+		void *buf);
+
+/**
+ * @brief Sensor batch data flush complete event receive callback.
+ *
+ * @param handle The sensor instance handle.
+
+ */
+typedef void (*senss_flush_complete_t)(
+		int handle);
+
+
+/**
+ * @struct senss_sensor_info
+ * @brief Sensor basic constant information
+ *
+ */
+struct senss_sensor_info {
+	/** Name of the sensor instance */
+	const char *name;
+
+	/** Friendly name of the sensor instance */
+	const char *friendly_name;
+
+	/** Vendor name of the sensor instance */
+	const char *vendor;
+
+	/** Model name of the sensor instance */
+	const char *model;
+
+	/** Sensor type */
+	int32_t type;
+
+	/** Sensor index in the sensor info array returned from senss_get_sensors() */
+	int32_t sensor_index;
```

Comment:
Discussed previously, this will not scale to have the index defined by the devicetree. It makes collisions too likely when overlays come into play. We should be using the `chosen` nodes to set the default all others are just as devicetree lays them out in memory. So for example, calling `get_sensor_by_type_and_instance()` with 0 as the index, will return the `chosen` node, passing `1` will just get the second one of that type. This will always be the same number between reboots but may be different between builds.

Reply:
removed senso_index, and will use DEVICE_DT_GET(DT_NODELABEL()) to get sensor device